pulse2percept/utils/geometry.py

```diff
@@ -221,24 +283,110 @@ def plot(self, transform=None, label=None, style='hull', autoscale=True,
 
 
 class VisualFieldMap(object, metaclass=ABCMeta):
-    """Base class for a visual field map (retinotopy)
+    """ Base template class for a visual field map (retinotopy) """
 
-    A template
+    @abstractmethod
+    def region_mappings(self):
+        """ Returns a dict containing the region(s) that this retinotopy maps 
+            to, and the corresponding mapping function(s).
+        """
+        raise NotImplementedError
 
-    """
+    @abstractmethod
+    def inv_region_mappings(self):
+        """ Returns a dict containing the region(s) that this retinotopy maps 
+            from, and the corresponding inverse mapping function(s).
+        """
+        raise NotImplementedError
 
+
+class RetinalMap(VisualFieldMap):
+    """ Template class for retinal visual field maps, which only have 1 region."""
+    @classmethod
+    def region_mappings(cls):
+        return {'ret' : cls.dva2ret}
+    
+    @classmethod
+    def inv_region_mappings(cls):
+        return {'ret' : cls.ret2dva}
+    
+    @staticmethod
     @abstractmethod
```

Comment:
Not completely sure why all the retinal retinotopy transforms are static. Models always instantiate them anyways. Cortical transforms require state so this pattern doesn't work there